connector/interface.go

```diff
@@ -60,6 +60,11 @@ type ConnectorConfig interface {
 	Connector(ns url.URL, loginFunc oidc.LoginFunc, tpls *template.Template) (Connector, error)
 }
 
+// Groups is an optional interface implemented by a Connector.
+type Groups interface {
+	Groups(fullUserID string) ([]string, error)
```

Comment:
Actually, thinking that through more: it _seems_ better to just add this method to the existing interface, and instead make other connectors which do not support grouping return an error, or something. What do you think?